recipes/leptonica/all/conanfile.py

```diff
@@ -50,106 +76,116 @@ def requirements(self):
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_PNG'] = not self.options.with_png
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_TIFF'] = not self.options.with_tiff
-        cmake.definitions['CMAKE_DISABLE_FIND_PACKAGE_JPEG'] = not self.options.with_jpeg
-
-        cmake.definitions['SW_BUILD'] = False
-
-        cmake.configure(source_folder=self._source_subfolder)
-        return cmake
-
-    def build(self):
-        cmake_original = os.path.join(self._source_subfolder,
-                                      "CMakeListsOriginal.txt")
-        # disable pkgconfig
-        tools.replace_in_file(cmake_original,
-                              "if (PKG_CONFIG_FOUND)",
-                              "if (FALSE)")
-
-        # short out conditionals that don't respect
-        # CMAKE_DISABLE_FIND_PACKAGE_x
-        if not self.options.with_webp:
-            tools.replace_in_file(cmake_original,
-                                  "if(NOT WEBP)",
-                                  "if(FALSE)")
-            tools.replace_in_file(cmake_original,
-                                  "if(NOT WEBPMUX)",
-                                  "if(FALSE)")
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+        cmakelists_src = os.path.join(self._source_subfolder, "src", "CMakeLists.txt")
+        cmake_configure = os.path.join(self._source_subfolder, "cmake", "Configure.cmake")
+
+        # Fix installation
+        tools.replace_in_file(cmakelists_src, "${CMAKE_BINARY_DIR}", "${PROJECT_BINARY_DIR}")
+
+        # Honor options
+        if not self.options.with_zlib:
+            tools.replace_in_file(cmakelists_src, "if (ZLIB_LIBRARIES)", "if(0)")
+            tools.replace_in_file(cmake_configure, "if (ZLIB_FOUND)", "if(0)")
+        if not self.options.with_gif:
+            tools.replace_in_file(cmakelists_src, "if (GIF_LIBRARIES)", "if(0)")
+            tools.replace_in_file(cmake_configure, "if (GIF_FOUND)", "if(0)")
+        if not self.options.with_jpeg:
+            tools.replace_in_file(cmakelists_src, "if (JPEG_LIBRARIES)", "if(0)")
+            tools.replace_in_file(cmake_configure, "if (JPEG_FOUND)", "if(0)")
+        if not self.options.with_png:
+            tools.replace_in_file(cmakelists_src, "if (PNG_LIBRARIES)", "if(0)")
+            tools.replace_in_file(cmake_configure, "if (PNG_FOUND)", "if(0)")
+        if not self.options.with_tiff:
+            tools.replace_in_file(cmakelists_src, "if (TIFF_LIBRARIES)", "if(0)")
+            tools.replace_in_file(cmake_configure, "if (TIFF_FOUND)", "if(0)")
         if not self.options.with_openjpeg:
-            tools.replace_in_file(cmake_original,
-                                  "if(NOT JP2K)",
-                                  "if(FALSE)")
+            tools.replace_in_file(cmakelists_src, "if (JP2K_FOUND)", "if(0)")
```

Comment:
Unfortunately, it is not enough for openjpeg/webp when pkgconfig is used.

Even when we unset `HAVE_LIBJP2K` in these branches and omit JP2K_INCLUDE_DIRS/JP2K_LIBRARIES from target, `HAVE_LIBJP2K` will be set in root `CMakeLists.txt` when pkgconfig succeeds with finding system openjpeg. One more patch is needed to replace `if(NOT JP2K)`. I can reproduce it only when a mentioned patch to `openjpeg-2.4`.

The same thing with WEBP.

Reply:
Good catch, I'll fix that.